site/content/blog/2019-10-21-localizing-matterpoll/index.md

```diff
@@ -0,0 +1,188 @@
+
+This change is very handy for both developers and contributors and the main reason we ended up choosing v2 for Matterpoll. Other notable changes are:
+
+- Better support for plural forms with options for e.g., `Zero`, `One`, `Few` and `Many`.
+- No global state, which helps with running tests in parallel.
+- Use of [`golang.org/x/text/language`](https://godoc.org/golang.org/x/text/language) for standardized behavior.
+
+For further information refer to [Changelog](https://github.com/nicksnyder/go-i18n/blob/master/CHANGELOG.md#v2) of go-i18n.
+
+Also, it's a good idea to stay up to date with the major version of the libraries that you are using. This ensures you get the latest bug fixes because open libraries don't backport fixes.
+
+These improvements are why we decided to go with v2 of go-i18n.
+
+
+#### Choosing a File Format
+
+Go-i18n supports multiple file formats for the translation file. In fact, you can use any file format as long as you implement an [unmarshal function](https://godoc.org/github.com/nicksnyder/go-i18n/v2/i18n#UnmarshalFunc) yourself. The most common formats are `JSON`, `TOML` and `YAML`.
+
+What is the preferred format for us? `goi18n` uses `TOML` by default, but since the Mattermost server uses `JSON` we decided to stay with that format. We could just as easily have stuck with the default format 'TOML' of 'goi18n' for consistency with the library default with no adverse effect.
+
+
+### Integrating go-i18n into Matterpoll
+
+With these decisions made we started working on integrating go-i18n into Matterpoll. But first we have to introduce three concepts that go-i18n uses (technically speaking they are just structs):
+
+- [`Bundle`](https://godoc.org/github.com/nicksnyder/go-i18n/v2/i18n#Bundle): A `Bundle` contains the translations.
+- [`Message`](https://godoc.org/github.com/nicksnyder/go-i18n/v2/i18n#Message): A `Message` is a translation string and can contain plural rules.
+- [`Localizer`](https://godoc.org/github.com/nicksnyder/go-i18n/v2/i18n#Localizer): A `Localizer` translates `Messages` to a specific language using a `Bundle`.
+
+Because Matterpoll needs to fetch the translation files on startup, they need to be included in the plugin bundle. The Makefile of the plugin starter template allows plugin developers to simply place their file into `assets` [to get them included in the bundle](https://github.com/mattermost/mattermost-plugin-starter-template#how-do-i-include-assets-in-the-plugin-bundle). 
```

Comment:
```suggestion
Because Matterpoll needs to fetch the translation files on startup, they need to be included in the plugin bundle. The `makefile` of the plugin starter template allows plugin developers to simply place their file into `assets` [to get them included in the bundle](https://github.com/mattermost/mattermost-plugin-starter-template#how-do-i-include-assets-in-the-plugin-bundle). 
```

Reply:
"Makefile" is a specific term.